test/e2e-common.sh

```diff
@@ -106,24 +110,32 @@ function pubsub_setup() {
 }
 
 # Create resources required for GCP Broker authentication setup.
-function gcp_broker_setup() {
+function broker_auth_setup() {
   echo "Authentication setup for GCP Broker"
   local auth_mode=${1}
 
   if [ "${auth_mode}" == "secret" ]; then
-    kubectl -n "${CONTROL_PLANE_NAMESPACE}" create secret generic "${GCP_BROKER_SECRET_NAME}" --from-file=key.json="${PUBSUB_SERVICE_ACCOUNT_KEY_TEMP}"
+    if (( ! IS_PROW )); then
```

Comment:
Can you add a comment here explaining why this `if` statement was necessary?

Reply:
Now that we use a different GSA for the broker we have to create it and grant the right permissions, just similar to the `sources_auth_setup`. I can add a comment similar to the one there:

```# When not running on Prow we need to set up a service account for broker.```